tests/migrations/test_autodetector.py

```diff
@@ -2181,6 +2181,107 @@ def test_rename_model_reverse_relation_dependencies(self):
             changes, "testapp", 0, 0, old_name="EntityA", new_name="RenamedEntityA"
         )
 
+    def test_move_model(self):
+        before = [self.author_name, self.book]
+        after = [
+            ModelState(
+                "otherapp",
+                "Author",
+                [
+                    ("id", models.AutoField(primary_key=True)),
+                    ("name", models.CharField(max_length=200)),
+                ],
+                options={"db_table": "testapp_author"},
+            ),
+            ModelState(
+                "otherapp",
+                "Book",
+                [
+                    ("id", models.AutoField(primary_key=True)),
+                    ("author", models.ForeignKey("otherapp.Author", models.CASCADE)),
+                    ("title", models.CharField(max_length=200)),
+                ],
+            ),
+        ]
+        changes = self.get_changes(
+            before, after, MigrationQuestioner({"ask_move_model": True})
+        )
+        self.assertNumberMigrations(changes, "otherapp", 2)
+
+        self.assertMigrationDependencies(changes, "otherapp", 0, [])
+        self.assertOperationTypes(changes, "otherapp", 0, ["CreateModel"])
+        self.assertMigrationDependencies(
+            changes, "otherapp", 1, [("otherapp", "auto_1"), ("testapp", "auto_1")]
+        )
+        self.assertOperationTypes(
+            changes, "otherapp", 1, ["AlterModelOptions", "AlterField"]
+        )
+
+        self.assertNumberMigrations(changes, "testapp", 2)
+
+        self.assertMigrationDependencies(
+            changes, "testapp", 0, [("otherapp", "auto_1")]
+        )
+        self.assertOperationTypes(changes, "testapp", 0, ["AlterModelOptions"])
+        self.assertMigrationDependencies(
+            changes, "testapp", 1, [("otherapp", "auto_2"), ("testapp", "auto_1")]
+        )
+        self.assertOperationTypes(changes, "testapp", 1, ["DeleteModel"])
+
+    def test_move_unmanaged_model(self):
```

Comment:
Not sure it's worth considering this case as pointed out above.

When dealing with unmanaged models the currently generated `DeleteModel` and `CreateModel` operations are just fine as they don't result in any changes to the database; it's the whole premise of this approach for managed models.